services/utils.js

```diff
@@ -32,36 +28,29 @@ const compare = async (storedPassword, userSalt, input, peperRange) => {
   return false;
 };
 
-const validFields = [GRAPH_FIELDS.WEIGHT, GRAPH_FIELDS.DESTINATION];
-
-const objIsEdge = (obj, vertexIndex) => {
-  const fields = Object.keys(obj);
-  const isValidFieldsLength = fields.length == validFields.length;
-
-  const hasDestinationKey = fields.includes(GRAPH_FIELDS.DESTINATION);
-  const hasDestinationValue = obj[GRAPH_FIELDS.DESTINATION] !== null;
-  const isConnectedToSelf = obj[GRAPH_FIELDS.DESTINATION] === `${vertexIndex}`;
-
-  const isValidDestination =
-    hasDestinationKey && hasDestinationValue && !isConnectedToSelf;
-
-  const hasWeightField = fields.includes(GRAPH_FIELDS.WEIGHT);
-  const isNumericWeight = !isNaN(parseFloat(obj[GRAPH_FIELDS.WEIGHT]));
-
-  const isValidWeight = hasWeightField && isNumericWeight;
-
-  return isValidFieldsLength && isValidWeight && isValidDestination;
+const validateList = (items, getKey) => {
+  const itemMap = new Map();
+  for (const item of items) {
+    const key = getKey(item);
+    if (itemMap.has(key)) {
+      throw new Error(`Duplicate item found: ${key}`);
+    } else {
+      itemMap.set(key, true);
+    }
+  }
```

Comment:
I don`t think that you need the else in here (If you throw an error the function is ending at that point) 

Do you think that something like this will work : 


```js
const validateList = (items, getKey) => {
  const itemMap = new Map();
 let key;
  for (const item of items) {
    key = getKey(item);
    if (itemMap.has(key)) throw new Error(`Duplicate item found: ${key}`);
    itemMap.set(key, true);
  }
};
```

Reply:
yes

Reply:
fixed